### PR TITLE
Add ApplePayPayload interface

### PR DESCRIPTION
**Pull request by ONE001**
Add missing payload interface for ApplePay tokenize method

https://braintree.github.io/braintree-web/current/ApplePay.html#~tokenizePayload

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/braintree-web/modules/apple-pay.d.ts [hunk 1]
```diff
@@ -64,10 +64,32 @@ export enum ApplePayStatusCodes {
...
+    commercial: ApplePayTokenizeValues;
+        countryOfIssuance: string;
+        debit: ApplePayTokenizeValues;
+        durbinRegulated: ApplePayTokenizeValues;
+        healthcare: ApplePayTokenizeValues;
+        issuingBank: ApplePayTokenizeValues;
+        payroll: ApplePayTokenizeValues;
+        prepaid: ApplePayTokenizeValues;
+        productId: string;
+    };
```
Comment (jbuckner) [inline types/braintree-web/modules/apple-pay.d.ts:92]: Interesting.. did the payload change or was it just wrong before?

File: types/braintree-web/modules/apple-pay.d.ts [hunk 2]
```diff
@@ -64,10 +64,32 @@ export enum ApplePayStatusCodes {
     STATUS_PIN_LOCKOUT,
 }
 
+export type ApplePayTokenizeValues = 'Yes' | 'No' | 'Unknown';
+
+export interface ApplePayDetails {
+    cardType: string;
+    cardholderName: string;
+    dpanLastTwo: string;
+    paymentInstrumentName: boolean;
```
Comment (jbuckner) [inline types/braintree-web/modules/apple-pay.d.ts:73]: I didn't see `paymentInstrumentName` in the docs. Is it undocumented?

File: types/braintree-web/modules/apple-pay.d.ts [hunk 3]
```diff
@@ -64,10 +64,32 @@ export enum ApplePayStatusCodes {
...
-    merchantIdentifier: string;
-    domainName: string;
-    displayName: string;
+    nonce: string;
+    description: string;
+    type: string;
+    consumed: boolean;
+    details: ApplePayDetails;
+    binData: {
+    commercial: ApplePayTokenizeValues;
```
Comment (jbuckner) [inline types/braintree-web/modules/apple-pay.d.ts:83]: Minor, but could you indent this to match the others?

**Review by jbuckner — COMMENTED** [review]
Thank you for the update! Some questions and one suggestion, but overall looks good!

File: types/braintree-web/modules/apple-pay.d.ts (hunk 1 above)
Comment (ONE001) [reply types/braintree-web/modules/apple-pay.d.ts:92]: It was wrong from the beginning. It wasn't used anywhere and didn't have nonce at all... So, I digged deeper and noticed that it is completely wrong...

File: types/braintree-web/modules/apple-pay.d.ts (hunk 2 above)
Comment (ONE001) [reply types/braintree-web/modules/apple-pay.d.ts:73]: yep, I took it from real response, not from docs... I'll remove it, so it would match with the doc

**Review by jbuckner — APPROVED** [review]
Solid! Thanks for the update!

**Comment by ONE001** [comment]
Ready to merge

**Pull request merged**